medicines/doc-index-updater/src/delete_manager/mod.rs

```diff
@@ -108,6 +107,9 @@ pub async fn process_message(message: DeleteMessage) -> Result<Uuid, ProcessMess
         &storage_container_name
```

Comment:
I believe this will error if the blob has already been deleted? So there will still be an error if the first attempt deletes the blob but fails to delete from the index and then it's retried.

Reply:
Good point. I'll look into it.

Reply:
Best way to achieve consistent response is probably to treat deleting blob & deleting index as separate operations:

- Receive message to delete document
- Look up in index
- Delete blob
- If successful, dispatch message to delete in index
- Receive message to delete index
- Delete index
- Mark as done

Thoughts?

Reply:
@TimboTambo Just to confirm, is the desired behaviour here:

 * if blob exists, delete it and remove it from index
 * if blob does not exist, emit a warning and remove if from index
 * if an error* occurs, raise an error and retry

?

<small>* — an error which isn't "blob does not exist"; for example, a the blob has an open lease, a permission error with the index, etc.</small>

Reply:
Agree that a more robust way of making idempotent would be to split out into two separate tasks with another message. As a short-term fix, I think @craiga's mentioned flow would be sufficient to prevent the observed issue from re-occurring.

Reply:
I'm not sure what log behavior is expected, but Accenture are expecting an **error** response if they get the job and the blob doesn't exist. I'm not sure that we need to give them the same if the index couldn't be deleted.

Reply:
Unit of Work pattern is the normal way of achieving atomicity across different services (index and blob) isn't it? treating as separate seems a mistake as yes you won't get errors, you'll just get inconsistencies.